docs/details/other-components/file_explorer.rst

```diff
@@ -4,85 +4,183 @@
 
-.. _file_explorer_usage:
+When enabled, there is also a "Quick-Access" panel on the left, which provides a
+convenient way to reach parts of the file system that are frequently accessed.
+Available "Quick-Access" destinations are:
 
-Usage
------
+- File System,
+- HOME,
+- Video,
+- Pictures,
+- Music, and
+- Documents.
 
-Enable :c:macro:`LV_USE_FILE_EXPLORER` in ``lv_conf.h``.
+You specify what paths these lead to during ``lv_file_explorer``\ 's initialization.
 
-First use :cpp:expr:`lv_file_explorer_create(lv_screen_active())` to create a file
-explorer, The default size is the screen size. After that, you can
-customize the style like widget.
+``lv_file_explorer`` only provides the file browsing and events caused by user
+activity (e.g. clicking a file), but does not provide the actual file operations.
+Client code must hook various events and decide what to do when they are emitted
+(e.g. a click or double-click on a file).  The actions taken might to open the file,
+display it, send it to some other part of the application, etc..
+``lv_file_explorer`` passes the full path and name of file that was clicked to the
+event callback functions.  What happens next is up to the application designer.
 
-Quick access
-~~~~~~~~~~~~
+``lv_file_explorer`` uses the :ref:`lv_table` Widget for the "Browsing Area", and the
+:ref:`lv_list` Widget for the "Quick-Access" panel when it is enabled.  Thus,
+:c:macro:`LV_USE_TABLE` macro must be set to a non-zero value in ``lv_conf.h`` in
+order to use ``lv_file_explorer``, and and :c:macro:`LV_USE_LIST` must be set to a
+non-zero value to use the "Quick-Access" panel.
 
-The quick access bar is optional. You can turn off
-:c:macro:`LV_FILE_EXPLORER_QUICK_ACCESS` in ``lv_conf.h`` so that the quick
-access bar will not be created. This can save some memory, but not much.
-After the quick access bar is created, it can be hidden by clicking the
-button at the top left corner of the browsing area, which is very useful
-for small screen devices.
+.. note::
+
+    In order to use File Explorer, :ref:`overview_file_system` has to be set up and
+    know about all the drive letters you use when passing paths to File System
+    (described below).
+
+
+
+Prerequisites
+*************
+
+If you haven't already done so, you will need to learn about the LVGL :ref:`File
+System abstraction <overview_file_system>`, since it must be set up and be functional
+for File Explorer to work.
+
+
+
+.. _file_explorer_usage:
+
+Usage
+*****
+
+Set :c:macro:`LV_USE_FILE_EXPLORER` to a non-zero value in ``lv_conf.h``.
+
+First use :cpp:expr:`lv_file_explorer_create(lv_screen_active())` to create a File
+Explorer.  The default size is the screen size.  After that, you can
+customize the style like any Widget.
+
+The size of the ``current_path`` buffer is set by :c:macro:`LV_FILE_EXPLORER_PATH_MAX_LEN`
+in ``lv_conf.h``.
+
+The object hierarchy of a freshly-created File Explorer looks like this:
+
+.. parsed-literal::
```

Comment:
It seems the [preview of the docs](http://crystal-clear-research.com/for_gabor/demo/docs.lvgl.io/master/details/other-components/file_explorer.html) is not updated with this. Is it correct or am I missing something?

Reply:
> It seems the [preview of the docs](http://crystal-clear-research.com/for_gabor/demo/docs.lvgl.io/master/details/other-components/file_explorer.html) is not updated with this. Is it correct or am I missing something?

You are correct!  It should be up to date now.  I don't know if it was just that one or not, but I updated all 4 "just in case".  :-)

Reply:
Thank you! What do you think about a more conventional way of describing it? E.g.
-   **File Explorer**: occupies full area of parent Widget, typically a Screen (Flex-Flow COLUMN)
    - **Container**  occupies full area of File Explorer (Flex grow 1) 
       - **Quick-Access Panel** left 22% of container (only created if enabled, Flex-Flow COLUMN)
          - **Device List** ..
            - ...
         - **Places List** ...
           - ...
      - **Browser Panel** ...

Reply:
@kisvegabor Done!